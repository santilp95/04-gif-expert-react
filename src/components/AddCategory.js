import React, { useState } from 'react'
import PropTypes from 'prop-types';

// llamar el componente atravez d destructurizacion
export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e)=>{
        // con esto se caputra el valor
        const valor= e.target.value;
        setInputValue(valor);
    }
    // recibe el event y se lo mando a algo ene ste ejemplo a form
    const handleSubmit = (e)=>{
        // para prevenir el comportamiento por defecto del formulario(osea que actualice toda la pagina)
        e.preventDefault();
        
        if(inputValue.trim().length>2){
            // recibe las categorias del componente padre e isnerta con el inputValue la nuestra
            setCategories(categories=>[inputValue,...categories]);
            setInputValue('');
        }

        
       
    }
    
    return (
        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <input 
                type="text"
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>
    )
}

// hacer obligatorio mandar la funcion
AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
