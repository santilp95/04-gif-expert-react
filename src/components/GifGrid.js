import React, { useState, useEffect } from 'react'

export const GifGrid = ({category}) => {

    const [count, setCount] = useState(0);

    // esto funciona para ejecutar cuando el compnente se rendiriza por primera vez si se manda el array vacio
    useEffect(() => {
        getGifs();
    },[])
        

    const  getGifs = async() =>{

        const url = `https://api.giphy.com/v1/gifs/search?q=JOJO&limit=10&api_key=qsUmBiU0pSR30iCqtxxSlrEeVjYfJAGP`;
        const respuesta = await fetch(url);
        const {data} = await respuesta.json();

        const gifs = data.map(img =>{
            return {
                id: img.id,
                title : img.title,
                url: img.images?.downsized_medium.url
            }
        })

        console.log(gifs);
    }

    

    return (
        <div>
            <h3>{category}</h3>
        </div>
    )
}
