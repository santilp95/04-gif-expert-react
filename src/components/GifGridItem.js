import React from 'react'

export const GifGridItem = ({title,url}) => {

    
    return (
        <div>
            <img src={url} alt={title} />
        </div>
    )
}
