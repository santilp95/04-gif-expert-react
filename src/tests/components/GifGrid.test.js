import'@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
describe('Pruebas componentes <GifGrid />', () => {
    const category = 'luna';
    let wrapper = shallow(<GifGrid category={category}/>);

    test('debe de mostrar <GifGridItem/> correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
});
