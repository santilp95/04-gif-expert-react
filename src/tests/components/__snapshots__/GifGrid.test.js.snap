// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas componentes <GifGrid /> debe de mostrar <GifGridItem/> correctamente 1`] = `
<Fragment>
  <h3>
    luna
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
